community/codegen/src/main/java/org/neo4j/codegen/ByteCodeUtils.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.codegen;
+
+import java.util.List;
+
+public final class ByteCodeUtils
```

Comment:
A lot of the functionality in this class already exists in `org.objectweb.asm.Type`.
It might still make sense to have this class, in order to own our interfaces, but the implementation could perhaps. be made by delegating to `org.objectweb.asm.Type`.